pinot-controller/pom.xml

```diff
@@ -166,5 +167,41 @@
         <npm.script>build</npm.script>
       </properties>
     </profile>
+    <profile>
+      <id>pinot-fastdev</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>build-helper-maven-plugin</artifactId>
+            <version>3.6.0</version>
+            <executions>
+              <execution>
+                <id>check-frontend-up-to-date</id>
+                <phase>process-sources</phase>
+                <goals>
+                  <goal>uptodate-property</goal>
+                </goals>
+                <configuration>
```

Comment:
I didn't know this plugin, it could be also useful in pinot-commons!

Reply:
This approach, while useful, isn't bulletproof. 
If one switches between branches then it won't trigger build if files in current branch are older than in the previous one. 
Still, it's better than checking whether target file exists.